src/cases/containers/restricted-case-access-container/restricted-case-access-container.component.ts

```diff
@@ -0,0 +1,72 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { of } from 'rxjs';
+import { switchMap, take } from 'rxjs/operators';
+import { CaseRole } from '../../../role-access/models/case-role.interface';
+import { AllocateRoleService } from '../../../role-access/services/allocate-role.service';
+import { Caseworker } from '../../../work-allocation/models/dtos';
+import { CaseworkerDataService } from '../../../work-allocation/services/caseworker-data.service';
+import { WASupportedJurisdictionsService } from '../../../work-allocation/services/wa-supported-jurisdiction.service';
+import { RestrictedCase } from '../../models/restricted-case.model';
+
+@Component({
+  selector: 'exui-restricted-case-access-container',
+  templateUrl: './restricted-case-access-container.component.html'
+})
+export class RestrictedCaseAccessContainerComponent implements OnInit {
+
+  public caseId: string;
+  public caseRoles: CaseRole[];
+  public caseWorkers: Caseworker[];
+  public idamIds: string[];
+  public restrictedCases: RestrictedCase[];
+
+  constructor(private readonly route: ActivatedRoute,
+              private readonly allocateService: AllocateRoleService,
+              private readonly caseworkerDataService: CaseworkerDataService,
+              private waSupportedJurisdictionsService: WASupportedJurisdictionsService) {
+  }
+
+  public ngOnInit(): void {
+    this.caseId = this.route.snapshot.params.cid;
+    this.allocateService.getCaseAccessRolesByCaseId(this.caseId).pipe(
+      switchMap(caseRoles => {
+        this.caseRoles = caseRoles;
+        return of(this.getUniqueIdamIds());
+      }), take(1),
+      switchMap(() => {
+        return this.waSupportedJurisdictionsService.getWASupportedJurisdictions();
```

Comment:
@johnbenjamin-hmcts - single-line expression can be rewritten without `return` keyword or curly braces (concise body)

Reply:
Thank you @Daniel-Lam, I also managed to add in the error check.